### ISSUE TITLE
Current Updates destroy BBB instance

### ISSUE THREAD

**Issue by etec-masterofsynapse** [bigbluebutton/bigbluebutton#19003]
**Describe the bug**
When I run `sudo apt update` on a production BBB instance, it currently shows these outstanding updates:

```
bbb-apps-akka/unknown 1:2.7-48 all [upgradable from: 1:2.7-44]
bbb-config/unknown 1:2.7-28 amd64 [upgradable from: 1:2.7-27]
bbb-etherpad/unknown 1:2.7-12 amd64 [upgradable from: 1:2.7-11]
bbb-html5/unknown 1:2.7-184 amd64 [upgradable from: 1:2.7-164]
bbb-learning-dashboard/unknown 1:2.7-11 amd64 [upgradable from: 1:2.7-10]
bbb-libreoffice-docker/unknown 1:2.7-5 amd64 [upgradable from: 1:2.7-2]
bbb-record-core/unknown 1:2.7-12 amd64 [upgradable from: 1:2.7-10]
bbb-web/unknown 1:2.7-40 amd64 [upgradable from: 1:2.7-34]
distro-info-data/focal-updates 0.43ubuntu1.14 all [upgradable from: 0.43ubuntu1.13]
libaom0/focal-updates,focal-security 1.0.0.errata1-3+deb11u1build0.20.04.1 amd64 [upgradable from: 1.0.0.errata1-3build1]
mongodb-database-tools/focal 100.9.0 amd64 [upgradable from: 100.8.0]
nodejs/nodistro 18.18.2-1nodesource1 amd64 [upgradable from: 18.18.1-1nodesource1]
```

If I install these via `sudo apt upgrade`, it destroys several BBB functions, especially the WebSocket proxy needed for Firefox Audio.
Also, `bbb-conf --check` shows these new errors:

```
# Warning: You have this server defined for https, but in
#
#   /usr/share/bigbluebutton/nginx/sip.nginx
#
#  did not find the use of https in definition for proxy_pass
#
#            proxy_pass http://x.x.x.x:5066;
#

# Warning: You have this server defined for https, but in
#
#   /usr/share/bigbluebutton/nginx/sip.nginx
#
#  did not find the use of port 7443 in definition for proxy_pass
#
#            proxy_pass http://x.x.x.x:5066;
#
```

And I noticed changed values in the `bbb-conf --check` output as well:

```
L23: ext-rtp-ip: $${local_ip_v4}
L24: ext-sip-ip: $${local_ip_v4}

L36: playback_host: bigbluebutton.example.com
L37: playback_protocol: http

L68: sipjsHackViaWs: false
```

These were the original values:

```
L23: ext-rtp-ip: $${external_rtp_ip}
L24: ext-sip-ip: $${external_sip_ip}

L36: playback_host: xx.xx.com
L37: playback_protocol: https

L68: sipjsHackViaWs: true
```

**Firewall and IP address type**
Public without NAT, own IPv4

**Comment by ffdixon** [comment]
Don't update BigBlueButton using `sudo apt-get update && sudo apt-get dist-upgrade`.  Instead, re-run the `bbb-install.sh` command again with the same parameters as when you originally ran it.

`bbb-install.sh` will run `sudo apt-get update && sudo apt-get dist-upgrade` and, more importantly, re-apply the necessary configuration so everything works again.

**Comment by marga-reflact** [comment]
When I have installed BBB on a server, I use the hold command so that no errors happen during apt update && apt upgrade -y. `apt-mark hold bbb-apps-akka``apt-mark hold bbb-config`.......

**Comment by etec-masterofsynapse** [comment]
@ffdixon @marga-reflact Thanks to both of you for your answers.
I wasn't sure if apt with an installed BBB was a good idea, but didn't find anything on the web or in the docs that discouraged or forbade it.
And the notice on apt-mark hold is very valuable, since cases exist where we would want to upgrade other packages without BBB being upgraded as well (auto-install security updates for example).

Can this hold procedure be implemented in the docs and can we also get a dedicated upgrade procedure page to make it absolutely clear what to do and what not to do?

**Comment by ffdixon** [comment]
Hi @etec-masterofsynapse, for upgrading 2.6 and 2.7, the docs recommend using `bbb-install.sh`

 https://docs.bigbluebutton.org./administration/install

We'll update the docs to make it clearer that just running `sudo apt-get update && sudo apt-get dist-upgrade` isn't sufficient.

**Comment by GhaziTriki** [comment]
Hi @etec-masterofsynapse I upgraded a server yesterday from 2.5 to 2.6 just by running `bbb-install.sh` https://docs.bigbluebutton.org/2.6/administration/install#upgrading-bigbluebutton-26 and it should be the same process for 2.7 also.

@ffdixon Adding a new `Before you upgrade` section to the docs to details what should be done (recordings, config...) and what mustn't be done can help avoiding such redundant issues. Your thoughts?

**Comment by etec-masterofsynapse** [comment]
@ffdixon @GhaziTriki Yes, the docs state the usage of bbb-install for a minor / major upgrade. But in my case, we had an "upgrade" from 2.7.1 to 2.7.2, which is neither a minor or major upgrade.

Therefore the dedicated page and the steps "Before the upgrade" still makes sense to me.

**Comment by mherrb** [comment]
Something could also be documented to setup apt so that regular OS package updates can be handled by cron-apt or unattended-upgrades without breaking bbb.  Not sure if the `apt mark hold` commands listed above are right (as they will also block the upgrades run by `bbb-install.sh`.

**Comment by lawilog** [comment]
I'm sorry but the proposed solutions are not practical. On a server facing the internet, you need to have automatic updates enabled for basic security reasons. Therefore, in all real world scenarios, some equivalent of "`apt update && apt upgrade`" will likely be run every night. If really `bbb-install.sh` should be re-run with the original parameters, then save those parameters in /etc and configure the deb-packages to re-run after update. (Or something like that.) https://www.debian.org/doc/debian-policy/ch-maintainerscripts.html
